### PR TITLE
finished readme, left off find_or_create_by for separate reading

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
needs :eyes:

@jmburges @AnnJohn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,284 @@
...
+Uh-oh, 99 Problems is off The Black Album, as we all know. Let's fix this. 
+
+```ruby
+ninety_nine_problems.album = "The Black Album"
+
+ninety_nine_problems.album
+# => "The Black Album"
+```
+
+Much better. Now we need to save this record back into the database. To do so, we'll need to use an `UPDATE` SQL statement. That statement would look something like this:
```
Comment (jmburges) [inline README.md:93]: So you wouldn't want to run a SQL statement with the `WHERE` clause against the name. There may be _many_ songs with the name "99 problems". The only unique identifier for that row is the ID. This example needs to be against an id

File: README.md [hunk 2]
```diff
@@ -1,3 +1,284 @@
...
+```sql
+UPDATE songs 
+SET album="The Black Album"
+WHERE name="99 Problems";
+```
+
+Let's put it all together using our SQLite3-Ruby gem magic. Remember, in this example, we assume our database connection is stored in `DB[:conn]`. 
+
+```ruby
+sql = "UPDATE songs SET album=#{ninety_nine_problems.album" WHERE name = ?"
```
Comment (jmburges) [inline README.md:104]: make it id based, not name based

Comment (jmburges) [inline README.md:104]: should have kept reading. JK

File: README.md (hunk 1 above)
Comment (jmburges) [inline README.md:93]: should have kept reading JK <3

File: README.md [hunk 3]
```diff
@@ -1,3 +1,284 @@
...
+
+Luckily for us, each song record in the songs table already has a unique value in the `id` column. That is the great thing about the primary key column––it is always unique. 
+
+Song records in the database table have a unique `id`, and our `Song` instances have an `id` attribute, but that attribute gets set to `nil` when an individual song is initialized. We want our `Song` instances to get assigned a unique `id` number, but when? and how?
+
+The unique `id` number of a `Song` instance should *come from the database*. When a song record gets inserted into the database, it automatically gets assigned a unique ID number. We need to grab that ID number *from the database record* and assign it to the `Song` instance's `id` attribute. 
+
+If that sounds confusing, check out this diagram:
+
+![](https://docs.google.com/drawings/d/1KYjce3m-ItD3j0hTuorBtlTgPoQSVg9v2sddGLVle8M/pub?w=958&h=497)
```
Comment (jmburges) [inline README.md]: put this image on S3

**Comment by jmburges** [comment]
few issues and :shipit:

File: README.md (hunk 3 above)
Comment (SophieDeBenedetto) [inline README.md]: @jmburges images on s3 not showing up. not sure why. It is in the s3 bucket, permission are "read: world".

**Pull request merged**